From a975cad25534e9f8db61b44159a43f947bf68bbd Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:07:56 -0400
Subject: [usb] xhci: Use pci_enable_msix_exact() instead of pci_enable_msix()

Message-id: <1410887463-20674-405-git-send-email-dzickus@redhat.com>
Patchwork-id: 95249
O-Subject: [RHEL7 PATCH 404/591] xhci: Use pci_enable_msix_exact() instead of pci_enable_msix()
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit a62445aead1484ea753496682ef8648a921479be
Author: Alexander Gordeev <agordeev@redhat.com>
Date:   Thu May 8 19:25:58 2014 +0300

    xhci: Use pci_enable_msix_exact() instead of pci_enable_msix()

    As result of deprecation of MSI-X/MSI enablement functions
    pci_enable_msix() and pci_enable_msi_block() all drivers
    using these two interfaces need to be updated to use the
    new pci_enable_msi_range()  or pci_enable_msi_exact()
    and pci_enable_msix_range() or pci_enable_msix_exact()
    interfaces.

    Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
    Cc: Sarah Sharp <sarah.a.sharp@linux.intel.com>
    Cc: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Cc: linux-usb@vger.kernel.org
    Cc: linux-pci@vger.kernel.org
    Signed-off-by: Mathias Nyman <mathias.nyman@linux.intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci.c b/drivers/usb/host/xhci.c
index 49709c8..1d08005 100644
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -291,7 +291,7 @@ static int xhci_setup_msix(struct xhci_hcd *xhci)
   xhci->msix_entries[i].vector = 0;
  }
 
- ret = pci_enable_msix(pdev, xhci->msix_entries, xhci->msix_count);
+ ret = pci_enable_msix_exact(pdev, xhci->msix_entries, xhci->msix_count);
  if (ret) {
   xhci_dbg_trace(xhci, trace_xhci_dbg_init,
     "Failed to enable MSI-X");
-- 
1.7.1